src/422where.js

```diff
@@ -15,28 +15,148 @@ yy.Select.prototype.compileWhere = function (query) {
+				var source = query.sources[rightIdx];
+				// Only set up optimization if not already done
+				if (!source.onleftfn) {
+					source.onleftfns = ls;
+					source.onrightfns = rs;
+					source.onleftfn = new Function('p,params,alasql', 'var y;return ' + ls);
+					source.onrightfn = new Function('p,params,alasql', 'var y;return ' + rs);
+					source.optimization = 'ix';
+				}
+			} else if (leftIdx > rightIdx) {
+				// leftAlias is the later source
+				var source = query.sources[leftIdx];
+				if (!source.onleftfn) {
+					source.onleftfns = rs;
+					source.onrightfns = ls;
+					source.onleftfn = new Function('p,params,alasql', 'var y;return ' + rs);
+					source.onrightfn = new Function('p,params,alasql', 'var y;return ' + ls);
+					source.optimization = 'ix';
+				}
+			}
+		} else if (leftAliases.length === 1 && rightAliases.length === 0) {
+			// Single-table condition (e.g., t1.a = 5)
+			var alias = leftAliases[0];
+			if (typeof aliasToIdx[alias] !== 'undefined') {
+				var source = query.sources[aliasToIdx[alias]];
+				source.srcwherefns = source.srcwherefns
+					? source.srcwherefns + '&&(' + ls + '==' + rs + ')'
+					: '(' + ls + '==' + rs + ')';
+			}
+		} else if (leftAliases.length === 0 && rightAliases.length === 1) {
+			// Single-table condition with alias on right
+			var alias = rightAliases[0];
+			if (typeof aliasToIdx[alias] !== 'undefined') {
+				var source = query.sources[aliasToIdx[alias]];
+				source.srcwherefns = source.srcwherefns
+					? source.srcwherefns + '&&(' + ls + '==' + rs + ')'
+					: '(' + ls + '==' + rs + ')';
+			}
+		}
+	});
+
+	// Compile the srcwherefn for sources that have single-table conditions
 	query.sources.forEach(function (source) {
 		if (source.srcwherefns) {
 			source.srcwherefn = new Function('p,params,alasql', 'var y;return ' + source.srcwherefns);
 		}
-		if (source.wxleftfns) {
-			source.wxleftfn = new Function('p,params,alasql', 'var y;return ' + source.wxleftfns);
+	});
+};
+
+// Helper function to extract all equality conditions from a WHERE clause
+function extractWhereConditions(where) {
+	var conditions = [];
+
+	function traverse(node) {
+		if (!node) return;
+
+		// Handle Expression wrapper - get the inner expression
+		if (node.expression) {
+			traverse(node.expression);
+			return;
 		}
-		if (source.wxrightfns) {
-			source.wxrightfn = new Function('p,params,alasql', 'var y;return ' + source.wxrightfns);
+
+		if (!(node instanceof yy.Op)) return;
+
+		if (node.op === 'AND') {
+			traverse(node.left);
+			traverse(node.right);
+		} else if (node.op === '=') {
+			conditions.push(node);
 		}
-		//		console.log(source.alias, source.wherefns)
-		//		console.log(source);
+	}
+
+	traverse(where);
+	return conditions;
+}
+
+// Helper function to extract table aliases from a JS expression
+function extractAliases(js) {
+	var matches = js.match(/p\['([^']+)'\]/g) || [];
+	var aliases = matches.map(function (m) {
+		return m.match(/p\['([^']+)'\]/)[1];
```

Comment:
Are you sure regex is the best way to find out? I feel like this invites to lots of potential errors when we rename the generated js somewhere else. Is there another way?

also, if you are ever to use regex, then please initialise outside of the operating function.